addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/RequestorJob.java

```diff
@@ -0,0 +1,182 @@
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.automation.jobs;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpSender;
+import org.zaproxy.addon.automation.AutomationEnvironment;
+import org.zaproxy.addon.automation.AutomationJob;
+import org.zaproxy.addon.automation.AutomationProgress;
+
+public class RequestorJob extends AutomationJob {
+
+    public static final String JOB_NAME = "requestor";
+    private HttpSender httpSender =
+            new HttpSender(
+                    Model.getSingleton().getOptionsParam().getConnectionParam(),
+                    true,
+                    HttpSender.MANUAL_REQUEST_INITIATOR);
+
+    public RequestorJob() {}
+
+    public RequestorJob(HttpSender httpSender) {
+        this.httpSender = httpSender;
+    }
+
+    @Override
+    public void verifyJobSpecificData(LinkedHashMap<?, ?> jobData, AutomationProgress progress) {
+        Object o = jobData.get("requests");
+        if (o == null) {
+            progress.error(
+                    Constant.messages.getString(
+                            "automation.error.requestor.norequests", this.getName()));
+            return;
+        }
+        if (o instanceof ArrayList<?>) {
+            ArrayList<?> requests = (ArrayList<?>) o;
+            for (Object request : requests) {
+                if (request instanceof LinkedHashMap<?, ?>) {
+                    LinkedHashMap<?, ?> requestMap = (LinkedHashMap<?, ?>) request;
+                    String url = (String) requestMap.get("url");
+                    try {
+                        new URI(url, true);
+                    } catch (URIException e) {
+                        progress.error(
+                                Constant.messages.getString(
+                                        "automation.error.requestor.badurl",
+                                        this.getName(),
+                                        url,
+                                        request));
+                    }
+                    String method = (String) requestMap.get("method");
```

Comment:
@thc202 suggested that there are other methods as well, and the user could make use of custom ones as well so we shouldn't restrict the method because of that.  https://wiki.owasp.org/index.php/Testing_for_HTTP_Methods_and_XST_(OWASP-CM-008)#Arbitrary_HTTP_Methods

Reply:
It could still default to GET, one less line for the common case.